projects/core/src/site-context/facade/base-site.service.ts

```diff
@@ -12,7 +12,7 @@ import { StateWithSiteContext } from '../store/state';
 import { SiteContext } from './site-context.interface';
 
 @Injectable()
-export class BaseSiteService implements SiteContext<string> {
+export class BaseSiteService implements SiteContext<BaseSite> {
```

Comment:
There are a number of breaking changes in this higher level API. I can however not oversee implications for other teams, let alone customer projects.

Reply:
I will create another service called `base-sites.service.ts`. So that in the new service ts, I can add all new functions there. This `base-site.service.ts ` will not be modified. So, there will be no breaking changes.

Reply:
Now, I combined baseSite with baseSites. We will handle breaking change in #9601